src/Compilers/CSharp/Portable/Symbols/MemberSignatureComparer.cs

```diff
@@ -686,5 +723,24 @@ private static bool IsVarargMethod(Symbol member)
         {
             return member.Kind == SymbolKind.Method && ((MethodSymbol)member).IsVararg;
         }
+
+        /// <summary>
+        /// Does the members match in terms of tuple names (both in their return type and parameters)?
+        ///
+        /// We'll look at the result of equality without tuple names (1) and with tuple names (2).
+        /// The question is whether there is a change in tuple element names only (3).
+        ///
+        /// member1                  vs. member2                   | (1) | (2) |    (3)    |
+        /// `(int a, int b) M()`     vs. `(int a, int b) M()`      | yes | yes |   match   |
+        /// `(int a, int b) M()`     vs. `(int x, int y) M()`      | yes | no  | different |
+        /// `void M((int a, int b))` vs. `void M((int x, int y))`  | yes | no  | different |
+        /// `int M()`                vs. `string M()`              | no  | no  |   match   |
+        ///
+        /// </summary>
+        internal static bool DifferOnlyInTupleNames(Symbol member1, Symbol member2)
```

Comment:
It looks like this method returns `true` if the members are distinct, even without tuples? If so, consider changing the name.